### ISSUE TITLE
Learning rate logging and argument names. 

### ISSUE THREAD

**Issue by vince62s** [OpenNMT/OpenNMT-py#280]
When using the noam scheme, it should not add a "decay at" starting at a certain epoch.
by default it starts decaying at epoch 9 but it should not since noam is effectively decaying already.

Quick fix could be to set lr_decay at 1 when using noam.

**Comment by JianyuZhan** [comment]
Hi, @vince62s , I am doing the `Optim` rewriting work to use `torch.optim.lr_scheduler`.  I agree with you, but during my testing, I found that our `noam` is updated based on the initial learning rate(`-learning_rate`, which defaults to be `1.0`), and the side effect of this it effectively ignore the `-start_decay_at` logic,  i.e.,  when `-start_decay_at` decays the lr from epoch 8(default) onward, the next `noam` update is not affected at all, since it calculates the factor, then multiply it with the initial learning rate, not current learning rate.

I think this behavior is broken, and I agree with you on this, I will fix it though.

**Comment by srush** [comment]
yeah, it just prints that, doesn't actually cause it to decay. but we need to fix these argument names and printing.

**Comment by vince62s** [comment]
Actually it would be great to print the learning rate when reporting INFO at each XX iteration.

**Comment by JianyuZhan** [comment]
@vince62s , It would be too noisy since `noam` is used at batch level(one epoch has several batches),  I will add in my code for debugging, but it is by default off, maybe I can add an option for user to toggle it on.

**Comment by vince62s** [comment]
if you think "noisy" in the way too many info on the log, I think you can gain like when reporting src and tgt tok /sec:
XXXX src; YYYY tgt; tok/s
FYI the Lua version report the LR
anyway an option is fine, but monitoring noam all the way through is kind of interesting.

**Comment by JianyuZhan** [comment]
Okay, and yes, It is interesting :-).

**Comment by srush** [comment]
I agree with @vince62s we should add this. We already do it when we report to Crayon for instance http://crayon.eastus.cloudapp.azure.com:8888/
